plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestThriftHttpMetastoreClient.java

```diff
@@ -0,0 +1,132 @@
+    private static final String httpToken = "test-token";
+    private final HiveMetastoreAuthentication noAuthentication = new NoHiveMetastoreAuthentication();
+    private final Duration timeout = new Duration(20, SECONDS);
+    private static InMemoryThriftMetastore delegate;
+
+    private static final String testDbName = "testdb";
+
+    private static class TestRequestHeaderInterceptor
+            implements MockThriftHttpMetastoreServer.TestHttpRequestInterceptor
+    {
+        private final Map<String, String> requestHeaders = new HashMap<>();
+
+        @Override
+        public void process(HttpServletRequest request)
+        {
+            requestHeaders.clear();
+            requestHeaders.put("key1", request.getHeader("key1"));
+            requestHeaders.put("key2", request.getHeader("key2"));
+            requestHeaders.put(HttpHeaders.AUTHORIZATION, request.getHeader(HttpHeaders.AUTHORIZATION));
+        }
+
+        public String getInterceptedHeader(String headerName)
+        {
+            return requestHeaders.get(headerName);
+        }
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public static void setup()
+            throws Exception
+    {
+        requestHeaderInterceptor = new TestRequestHeaderInterceptor();
+        File tempDir = Files.createTempDirectory(null).toFile();
+        tempDir.deleteOnExit();
+        delegate = new InMemoryThriftMetastore(new File(tempDir, "/metastore"), getHttpMetastoreConfig());
+        metastoreServer = new MockThriftHttpMetastoreServer(delegate, requestHeaderInterceptor);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public static void tearDown()
+            throws Exception
+    {
+        metastoreServer.stop();
+    }
+
+    private static ThriftMetastoreConfig getHttpMetastoreConfig()
+    {
+        ThriftMetastoreConfig config = new ThriftMetastoreConfig();
+        config.setTransportMode(ThriftMetastoreConfig.ThriftTransportMode.HTTP);
+        config.setHttpPath(MockThriftHttpMetastoreServer.httpPath);
+        config.setHttpToken(httpToken);
+        config.setAdditionalHeaders("key1=value1, key2=value2");
+        return config;
+    }
+
+    @Test
+    public void testHttpThriftConnection()
+            throws TException
+    {
+        Database db = new Database();
+        db.setName(testDbName);
+        delegate.createDatabase(db);
+
+        ThriftMetastoreClientFactory factory = new DefaultThriftMetastoreClientFactory(
+                Optional.empty(),
+                Optional.empty(),
+                timeout,
+                noAuthentication,
+                "localhost",
+                DefaultThriftMetastoreClientFactory.buildThriftHttpContext(getHttpMetastoreConfig()));
+        ThriftMetastoreClient client = factory.create(
+                HostAndPort.fromParts("localhost", metastoreServer.getPort()),
+                Optional.empty());
+        assertEquals(Lists.newArrayList(testDbName), client.getAllDatabases());
```

Comment:
Please use `AssertJ` for new tests

Reply:
My IDE could not find `AssertJ`. Did you mean `AssertJUnit`?